beacon-chain/sync/initial-sync/blocks_fetcher.go

```diff
@@ -137,6 +149,21 @@ func (f *blocksFetcher) loop() {
 		close(f.fetchResponses)
 	}()
 
+	// Periodically remove stale peer locks.
+	go func() {
+		ticker := time.NewTicker(peerLocksPollingInterval)
+		for {
+			select {
+			case <-ticker.C:
+				f.removeStalePeerLocks(peerLockMaxAge)
+			case <-f.ctx.Done():
+				ticker.Stop()
```

Comment:
Any reason to not return `ctx.Err()` here?

Reply:
My purpose of listening to `ctx.Done()` was to make sure we do not leak goroutines when fetcher is done. So, that return will simply wrap up surrounding goroutine and not leave it hanging - so, I am not sure what returning `ctx.Err()` will achieve here. Can you please elaborate more what should be done?

Reply:
Got it. It seems like the error case is not a valid concern here. Sounds good to me then